src/UI/class-admin-columns-analytics.php

```diff
@@ -0,0 +1,248 @@
+			return;
+		}
+
+		$key = $this->get_unique_stats_key_of_current_post();
+
+		if ( '' === $key || ! isset( $this->parsely_stats_map[ $key ] ) ) {
+			echo '-';
+			return;
+		}
+
+		$metrics = $this->parsely_stats_map[ $key ];
+
+		if ( isset( $metrics['views'] ) ) {
+			$views = $metrics['views'];
+
+			echo "<span class='parsely-post-page-views'>"
+				. esc_html( strval( $views ) ) . ' '
+				. esc_html( _n( 'page view', 'page views', $views, 'wp-parsely' ) )
+				. '</span> <br/>';
+		}
+
+		if ( isset( $metrics['visitors'] ) ) {
+			$visitors = $metrics['visitors'];
+
+			echo "<span class='parsely-post-visitors'>"
+				. esc_html( strval( $visitors ) ) . ' '
+				. esc_html( _n( 'visitor', 'visitors', $visitors, 'wp-parsely' ) )
+				. '</span> <br/>';
+		}
+
+		if ( isset( $metrics['avg_engaged'] ) ) {
+			$avg_engaged     = $metrics['avg_engaged'];
+			$engaged_minutes = (int) floor( $avg_engaged );
+
+			if ( 0 === $engaged_minutes ) {
+				$engaged_seconds = (int) ( $avg_engaged * 60 );
+			} else {
+				$engaged_seconds = (int) ( fmod( $avg_engaged, $engaged_minutes ) * 60 );
+			}
+
+			echo "<span class='parsely-post-avg_engaged'>"
+				. esc_html( strval( $engaged_minutes ) ) . ':' . esc_html( strval( $engaged_seconds ) ) . ' '
+				. esc_html__( 'avg time', 'wp-parsely' )
+				. '</span> <br/>';
+		}
+	}
+
+	/**
+	 * Get publish date params for analytics API.
+	 *
+	 * For getting date params we loop through all the posts and calculate
+	 * the min and max publish dates based on which we will limit the analytics API.
+	 *
+	 * @param WP_Post[] $posts Array of post objects.
+	 *
+	 * @return Analytics_Post_API_Params|null
+	 */
+	private function get_publish_date_params_for_analytics_api( array $posts ) {
+		if ( count( $posts ) === 0 ) {
+			return null;
+		}
+
+		$max_date_time = '';
+		$min_date_time = '';
+
+		foreach ( $posts as $post ) {
+			$published_date_time = $post->post_date;
+
+			if ( '' === $min_date_time || $published_date_time < $min_date_time ) {
+				$min_date_time = $published_date_time;
+			}
+
+			if ( $published_date_time > $max_date_time ) {
+				$max_date_time = $published_date_time;
+			}
+		}
+
+		$date_format = 'Y-m-d';
+
+		// TODO: Handle timezones if needed.
+		return array(
+			'pub_date_start' => ( new DateTime( $min_date_time ) )->format( $date_format ),
+			'pub_date_end'   => ( new DateTime( $max_date_time ) )->format( $date_format ),
+		);
+	}
+
+	/**
+	 * Get unique key which we can use for Parse.ly stats map.
+	 * Mainly we need this because Parse.ly doesn't have anything unique in its analytics API.
+	 *
+	 * @param Analytics_Post $analytics_post Post analytics obj returned from Parse.ly API.
+	 *
+	 * @return string
+	 */
+	private function get_unique_stats_key_from_analytics( $analytics_post ): string {
+		if ( ! isset( $analytics_post['title'] ) || ! isset( $analytics_post['pub_date'] ) ) {
+			return '';
+		}
+
+		return $analytics_post['title'] . '-' . $analytics_post['pub_date'];
```

Comment:
There is nothing unique in Parsely API so I have come up with this formula which is not 100% correct. Please do let me know if you have any better suggestion.

Due to this I have also moved #265 to current milestone to decrease the chances of mismatch between parsely api response and WP posts.